jelares/portfolio/src/main/webapp/javascript/parallax.min.js

```diff
@@ -0,0 +1,243 @@
+        isFresh: !1,
+        isBusy: !1,
+        setup: function() {
+          function s() {
+            if (p == i.pageYOffset) return i.requestAnimationFrame(s), !1;
+            p = i.pageYOffset, h.render(), i.requestAnimationFrame(s)
+          }
+          if (!this.isReady) {
+            var h = this, r = t(e), a = t(i), n = function() {
+              o.winHeight = a.height(), o.winWidth = a.width(),
+              o.docHeight = r.height(), o.docWidth = r.width()
+            }, l = function() {
+              var t = a.scrollTop(), i = o.docHeight - o.winHeight,
+                  e = o.docWidth - o.winWidth;
+              o.scrollTop = Math.max(0, Math.min(i, t)),
+              o.scrollLeft = Math.max(0, Math.min(e, a.scrollLeft())),
+              o.overScroll = Math.max(t - i, Math.min(t, 0))
+            };
+            a.on('resize.px.parallax load.px.parallax', function() {
+               n(), h.refresh(), o.isFresh = !1, o.requestRender()
+             }).on('scroll.px.parallax load.px.parallax', function() {
+              l(), o.requestRender()
+            }), n(), l(), this.isReady = !0;
+            var p = -1;
+            s()
+          }
+        },
+        configure: function(i) {
+          'object' == typeof i &&
+              (delete i.refresh, delete i.render, t.extend(this.prototype, i))
+        },
+        refresh: function() {
+          t.each(this.sliders, function() {
+            this.refresh()
+          }), this.isFresh = !0
+        },
+        render: function() {
+          this.isFresh || this.refresh(), t.each(this.sliders, function() {
+            this.render()
+          })
+        },
+        requestRender: function() {
+          var t = this;
+          t.render(), t.isBusy = !1
+        },
+        destroy: function(e) {
+          var s, h = t(e).data('px.parallax');
+          for (h.$mirror.remove(), s = 0; s < this.sliders.length; s += 1)
+            this.sliders[s] == h && this.sliders.splice(s, 1);
+          t(e).data('px.parallax', !1),
+              0 === this.sliders.length &&
+              (t(i).off(
+                   'scroll.px.parallax resize.px.parallax load.px.parallax'),
+               this.isReady = !1, o.isSetup = !1)
+        }
+      });
+  var h = t.fn.parallax;
+  t.fn.parallax = function(s) {
+    return this.each(function() {
+      var h = t(this), r = 'object' == typeof s && s;
+      this == i || this == e || h.is('body') ?
+          o.configure(r) :
+          h.data('px.parallax') ?
+          'object' == typeof s && t.extend(h.data('px.parallax'), r) :
+          (r = t.extend({}, h.data(), r),
+           h.data('px.parallax', new o(this, r))),
+          'string' == typeof s && ('destroy' == s ? o.destroy(this) : o[s]())
+    })
+  }, t.fn.parallax.Constructor = o, t.fn.parallax.noConflict = function() {
+    return t.fn.parallax = h, this
+  }, t(function() {
+    t('[data-parallax="scroll"]').parallax()
+  })
+}(jQuery, window, document);
```

Comment:
I think this is fine for this project, but wanted to point out that for production projects we'd try to incorporate libraries via a package manager (like maven, but it looks like maven not supported, which I'm guessing is why you downloaded this directly).

Reply:
Sure, I'll keep this in mind for the capstone